dashboard/azimuth_plot.py

```diff
@@ -12,77 +13,137 @@
     ax.grid(color='white', linestyle='-', linewidth=0.5)
-    plt.title(f'Skyplot for {region} at {time_point.strftime("%H:%M:%S on %d/%m/%Y")}',
-              color='white')
-
+    plt.title(f'Skyplot for {region} at {time_point.strftime("%d/%m/%Y %H:%M")}', color='white')
     for spine in ax.spines.values():
         spine.set_visible(True)
         spine.set_color('white')
         spine.set_linewidth(1.5)
+
+    return fig, ax, scatter, body_properties_map
+
+
+def update_plot(frame, celestial_df, scatter, body_properties_map, text_elements, ax, region):
+    '''Updates the skyplot with new data for each hour in the dataset'''
+    time_point = frame
+    current_df = celestial_df[celestial_df['time_point'] == time_point]
+    sizes = current_df['body_name'].map(lambda x: body_properties_map[x]['size']).values
+    sizes = np.interp(sizes, [min(prop['size'] for prop in body_properties_map.values()),
+                    max(prop['size'] for prop in body_properties_map.values())], [50, 1000])
+
+    colors = current_df['body_name'].map(lambda x: body_properties_map[x]['color'])
+
+    scatter.set_offsets(np.c_[current_df['azimuth'],
+                        np.radians(current_df['altitude'])])
+    scatter.set_color(colors)
+    scatter.set_sizes(sizes)
+
+    for text in text_elements:
+        text.remove()
+    text_elements.clear()
+
+    for i in range(len(current_df)):
+        text = plt.text(current_df['azimuth'].iloc[i],
+                        np.radians(current_df['altitude']).iloc[i],
+                        current_df['body_name'].iloc[i],
+                        fontsize=10, ha='right', va='bottom', color='white')
+        text_elements.append(text)
+    ax.set_title(f'Skyplot for {region} at {time_point.strftime("%d/%m/%Y %H:%M")}', color='white')
+
+
+def animate_skyplot(celestial_df: pd.DataFrame, region: str):
+    '''Animates the sky plot'''
+    time_points = celestial_df['time_point'].unique()
+    fig, ax, scatter, body_properties_map = sky_plot(
+        celestial_df, region, time_points[0])
+    text_elements = []
+    ani = animation.FuncAnimation(fig, update_plot, frames=time_points,
+                                  fargs=(celestial_df, scatter, body_properties_map,
+                                         text_elements, ax, region),
+                                  interval=1000, repeat=True)
     plt.show()
 
-def make_sky_plot(region:str,time:dt) -> plt.subplot:
-    '''Runs script to make skyplot'''
-    data = get_azimuth_data(region,time)
+def make_sky_plot(region: str, time: dt) -> plt.subplot:
+    '''Orchestrates the pipeline to create the skyplot'''
+    data = get_azimuth_data(region, time)
     clean_data = transform_azimuth_data(data)
-    sky_plot(clean_data,region,time)
+    animate_skyplot(clean_data, region)
 
 
 if __name__ == '__main__':
     load_dotenv()
-    make_sky_plot('Scotland',dt(2024,10,26,20,0,0))
+    make_sky_plot('Scotland', dt(2024, 10, 25, 0, 0, 0))
```

Comment:
Please implement all of this in assembly, thanks.

Reply:
Of course, I'll make a ticket